server/src/test/integration/org/apache/cassandra/sidecar/routes/StreamStatsIntegrationTest.java

```diff
@@ -0,0 +1,202 @@
+import org.apache.cassandra.testing.ConfigurableCassandraTestContext;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests the stream stats endpoint with cassandra container.
+ */
+@ExtendWith(VertxExtension.class)
+public class StreamStatsIntegrationTest extends IntegrationTestBase
+{
+    @CassandraIntegrationTest(numDataDirsPerInstance = 4, nodesPerDc = 5, network = true, buildCluster = false)
+    void streamStatsTest(VertxTestContext context, ConfigurableCassandraTestContext cassandraTestContext) throws InterruptedException
+    {
+
+        BBHelperDecommissioningNode.reset();
+        UpgradeableCluster cluster = cassandraTestContext.configureAndStartCluster(
+        builder -> builder.withInstanceInitializer(BBHelperDecommissioningNode::install));
+        IUpgradeableInstance node = cluster.get(5);
+        IUpgradeableInstance seed = cluster.get(1);
+
+        createTestKeyspace();
+        createTestTableAndPopulate();
+
+        startAsync("Decommission node" + node.config().num(),
+                   () -> node.nodetoolResult("decommission").asserts().success());
+        AtomicBoolean hasStats = new AtomicBoolean(false);
+        AtomicBoolean dataReceived = new AtomicBoolean(false);
+
+        // Wait until nodes have reached expected state
+        awaitLatchOrThrow(BBHelperDecommissioningNode.transientStateStart, 2, TimeUnit.MINUTES, "transientStateStart");
+
+        ClusterUtils.awaitRingState(seed, node, "Leaving");
+        BBHelperDecommissioningNode.transientStateEnd.countDown();
+
+        for (int i = 0; i < 20; i++)
+        {
+            startAsync("Request-" + i , () -> {
+                try
+                {
+                    streamStats(context, hasStats, dataReceived);
+                }
+                catch (Exception e)
+                {
+                    throw new RuntimeException(e);
+                }
+            });
+
+            Uninterruptibles.sleepUninterruptibly(200, TimeUnit.MILLISECONDS);
+        }
+
+        assertThat(hasStats).isTrue();
+        assertThat(dataReceived).isTrue();
+        context.completeNow();
+        context.awaitCompletion(2, TimeUnit.MINUTES);
+    }
+
+    private void streamStats(VertxTestContext context, AtomicBoolean hasStats, AtomicBoolean dataReceived) throws Exception
+    {
+        String testRoute = "/api/v1/cassandra/stats/stream";
+        testWithClient(context, client -> {
+            BBHelperDecommissioningNode.transientStateEnd.countDown();
+            client.get(server.actualPort(), "127.0.0.1", testRoute)
+                  .send(context.succeeding(response -> {
+                       assertRingResponseOK(response, sidecarTestContext, hasStats, dataReceived);
+                  }));
+        });
+    }
+
+    void assertRingResponseOK(HttpResponse<Buffer> response, CassandraSidecarTestContext cassandraTestContext,
```

Comment:
this is not really a ring response? Also cassandraTestContext is unused
```suggestion
    void assertRingResponseOK(HttpResponse<Buffer> response,
```